### DISCUSSION TITLE
Mpfr data type support

### DISCUSSION THREAD

**Discussion by EazyAhmed** [nlohmann/json#4258] [Ideas]
Doubles and floats have precision inaccuracies. If i want to store a mpfr based numeric how can configure the library to support this.

**Comment by chrysante** [comment]
Have you considered to use the libraries printing and parsing functions and then simply store the numbers as decimal strings?

**Reply by EazyAhmed** [reply]
I have seen the template class basic_json. I think that's where i should be approaching this. I don't know about the parsing functions

**Reply by chrysante** [reply]
What I'm trying to say is: use `mpfr_snprintf` or `mpfr_get_str` to convert the number to a string, then store the string into json. When reading the value again, use `mpfr_set_str` to convert the string to mpfr type.

**Reply by EazyAhmed** [reply]
{Number1: 566} for example
In this json, is the json library internally gonna store this as a double. I know i can store this as "566". I want to change the internal mechanism to convert these numbers to mpfr implicitly. These functions that you mentioned will be useful. Im just asking whether there is an easy way to rewrite this mechanism. I saw this template class where i can specify the type.

**Reply by chrysante** [reply]
Yes, you define the `from_json` and `to_json` functions in your namespace, as shown in the example in the documentation. You don't need to fiddle with `basic_json` for this.

**Reply by EazyAhmed** [reply]
In my namespace?

**Reply by chrysante** [reply]
Just ignore that part if you don't deal with namespaces

**Reply by EazyAhmed** [reply]
Documentation says those functions should be in the same namespace as the type or the global. So i should be putting it into the mpfrs namespace right. If i put in my own namespace will the function be found?

**Reply by chrysante** [reply]
Aren't all mpfr types in the global namespace since it's a C library? So just put them in the global namespace

**Reply by EazyAhmed** [reply]
Im using boost multiprcesion actually. It has an mpfr wrapper class. Has its own namespace.
